diracx-db/src/diracx/db/sql/utils/job_status.py

```diff
@@ -108,112 +119,110 @@ async def reschedule_job(
-        "RescheduleCounter": reschedule_counter,
     }
 
-    set_job_status_result = await set_job_status(
-        job_id,
-        {
+    for job_id in surviving_job_ids:
+        class_ad_job = job_jdls[job_id]
+        class_ad_req = ClassAd("[]")
+        try:
+            # NOT A DATABASE OPERATION
+            await job_db.checkAndPrepareJob(
+                job_id,
+                class_ad_job,
+                class_ad_req,
+                jobs_to_resched[job_id]["Owner"],
+                jobs_to_resched[job_id]["OwnerGroup"],
+                {"RescheduleCounter": jobs_to_resched[job_id]["RescheduleCounter"]},
+                class_ad_job.getAttributeString("VirtualOrganization"),
+            )
+        except SErrorException as e:
+            failed[job_id] = {"detail": str(e)}
+            # surviving_job_ids.remove(job_id)
+            continue
+
+        priority = class_ad_job.getAttributeInt("Priority")
+        if priority is None:
+            priority = 0
+
+        site_list = class_ad_job.getListFromExpression("Site")
+        if not site_list:
+            site = "ANY"
+        elif len(site_list) > 1:
+            site = "Multiple"
+        else:
+            site = site_list[0]
+
+        reqJDL = class_ad_req.asJDL()
+        class_ad_job.insertAttributeInt("JobRequirements", reqJDL)
+        jobJDL = class_ad_job.asJDL()
+        # Replace the JobID placeholder if any
+        jobJDL = jobJDL.replace("%j", str(job_id))
+
+        additional_attrs = {
+            "Site": site,
+            "UserPriority": priority,
+            "RescheduleTime": datetime.now(tz=timezone.utc),
+            "RescheduleCounter": jobs_to_resched[job_id]["RescheduleCounter"],
+        }
+
+        # set new JDL
+        jdl_changes[job_id] = jobJDL
+
+        # set new status
+        status_changes[job_id] = {
             datetime.now(tz=timezone.utc): JobStatusUpdate(
                 Status=JobStatus.RECEIVED,
                 MinorStatus=JobMinorStatus.RESCHEDULED,
                 ApplicationStatus="Unknown",
             )
-        },
+        }
+        # set new attributes
+        attribute_changes[job_id].update(additional_attrs)
+
+    # BULK STATUS UPDATE
+    # DATABASE OPERATION
+    set_job_status_result = await set_job_statuses(
+        status_changes,
         config,
         job_db,
         job_logging_db,
         task_queue_db,
         background_task,
-        additional_attributes=additional_attrs,
+        additional_attributes=attribute_changes,
     )
 
-    reqJDL = classAdReq.asJDL()
-    classAdJob.insertAttributeInt("JobRequirements", reqJDL)
-
-    jobJDL = classAdJob.asJDL()
-
-    # Replace the JobID placeholder if any
-    jobJDL = jobJDL.replace("%j", str(job_id))
+    # BULK JDL UPDATE
+    # DATABASE OPERATION
 
     # Update JDL (Should we be doing this here?)
-    result = await job_db.setJobJDL(job_id, jobJDL)
+    # DATABASE OPERATION
+    await job_db.setJobJDLsBulk(jdl_changes)
```

Comment:
I should check which rows were successfully updated in the previous operation and exclude unsuccessful updates from the JDL update statement.